File: cloud-bridge/backend/verify_matching.js
const axios = require('axios');

const BASE_URL = 'http://localhost:8080/api';

const user = {
    username: `verifier_${Date.now()}`,
    password: 'password123',
    email: `verifier_${Date.now()}@example.com`,
    role: 'ENTERPRISE'
};

async function run() {
    try {
        console.log("Waiting for backend...");
        // Simple health check or wait
        await new Promise(r => setTimeout(r, 2000));

        console.log(`Registering user ${user.username}...`);
        try {
            await axios.post(`${BASE_URL}/auth/register`, user);
        } catch (e) {
            // Ignore if already exists (unlikely with timestamp)
            console.log("Registration skipped or failed:", e.response?.data || e.message);
        }

        console.log(`Logging in as ${user.username}...`);
        const loginRes = await axios.post(`${BASE_URL}/auth/login`, {
            username: user.username,
            password: user.password
        });
        const token = loginRes.data.token;
        console.log("Login successful, token received.");

        console.log("Testing matching algorithm...");
        const matchPayload = {
            description: "需要通过人工智能技术进行早期疾病筛查的诊断工具"
        };
        
        const matchRes = await axios.post(`${BASE_URL}/matching/match`, matchPayload, {
            headers: { Authorization: `Bearer ${token}` }
        });

        console.log("\n✅ 匹配成功！返回结果摘要：");
        const matches = matchRes.data.matches || [];
        console.log(`- 匹配到的成果数量: ${matches.length}`);
        
        if (matchRes.data.aiGraph) {
            const nodeCount = matchRes.data.aiGraph.nodes ? matchRes.data.aiGraph.nodes.length : 0;
            console.log(`- 知识图谱数据: 已提取 (Nodes: ${nodeCount})`);
        } else {
            console.log(`- 知识图谱数据: ⚠️ 未提取`);
        }

        if (matches.length > 0) {
            console.log("\n首个匹配详情:");
            console.log(JSON.stringify(matches[0], null, 2));
            
            // Check specific scoring fields to verify optimization
            const firstMatch = matches[0];
            if (firstMatch.score > 0) {
                 console.log(`\n验证分数: ${firstMatch.score} (预期 > 0)`);
            } else {
                 console.warn(`\n警告: 分数为 0, 匹配算法可能未生效`);
            }
        } else {
            console.warn("\n警告: 未找到匹配项");
        }

    } catch (e) {
        console.error("\n❌ 测试失败:", e.response?.data || e.message);
        if (e.response?.status === 401) {
             console.error("可能是认证失败。");
        } else if (e.response?.status === 405) {
             console.error("方法不允许 (405)。请检查 URL 或请求方法。");
        } else if (e.response?.status === 500) {
             console.error("服务器内部错误 (500)。请检查后端日志。");
        }
        process.exit(1);
    }
}

run();

File: cloud-bridge/backend/verify_matching.py
import urllib.request
import urllib.error
import json
import time
import sys

BASE_URL = "http://localhost:8080/api"

def wait_for_backend(timeout=300):
    print("正在等待后端服务就绪...", end="", flush=True)
    start = time.time()
    while time.time() - start < timeout:
        try:
            req = urllib.request.Request(f"{BASE_URL}/matching/match", method='GET')
            try:
                with urllib.request.urlopen(req, timeout=2) as response:
                    pass
            except urllib.error.HTTPError as e:
                if e.code in [405, 200, 401]: 
                    print("\n后端服务已就绪！")
                    return True
        except Exception:
            pass
        time.sleep(2)
        print(".", end="", flush=True)
    print("\n等待超时，请检查 Docker 日志。")
    return False

def get_token():
    url = f"{BASE_URL}/auth/login"
    payload = {
        "username": "admin",
        "password": "admin123"
    }
    data = json.dumps(payload).encode('utf-8')
    req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
    
    print(f"\n正在尝试登录获取 Token: {url}")
    try:
        with urllib.request.urlopen(req) as response:
            result = json.loads(response.read().decode('utf-8'))
            token = result.get("token")
            if token:
                print("✅ 登录成功，已获取 Token")
                return token
    except Exception as e:
        print(f"❌ 登录失败: {e}")
    return None

def test_matching(token):
    url = f"{BASE_URL}/matching/match"
    payload = {
        "description": "需要通过人工智能技术进行早期疾病筛查的诊断工具",
        "domains": ["医疗", "AI"] 
    }
    
    data = json.dumps(payload).encode('utf-8')
    headers = {
        'Content-Type': 'application/json',
        'Authorization': f'Bearer {token}'
    }
    req = urllib.request.Request(url, data=data, headers=headers)
    
    print(f"\n正在发送测试请求: {url}")
    print(f"请求参数: {json.dumps(payload, ensure_ascii=False)}")
    
    try:
        with urllib.request.urlopen(req) as response:
            response_body = response.read().decode('utf-8')
            # print(f"原始响应: {response_body}")
            result = json.loads(response_body)
            print("\n✅ 匹配服务调用成功！")
            
            matches = result.get("matches", [])
            print(f"- 匹配到的成果数量: {len(matches)}")
            
            if "aiGraph" in result:
                 print(f"- 知识图谱数据: 已提取")
            else:
                 print(f"- 知识图谱数据: ⚠️ 未提取")
            
            if "relatedKeywords" in result:
                print(f"- 提取的关键词: {result.get('relatedKeywords')}")
                 
            if matches:
                print(f"\nTop 1 匹配结果: {matches[0].get('title')} (得分: {matches[0].get('score')})")
                
            return True
    except urllib.error.HTTPError as e:
        print(f"\n❌ 请求失败 (HTTP {e.code}): {e.read().decode('utf-8')}")
    except Exception as e:
        print(f"\n❌ 请求出错: {e}")
    return False

if __name__ == "__main__":
    if wait_for_backend():
        token = get_token()
        if token:
            test_matching(token)
        else:
            print("无法继续测试：未获得有效 Token")
